### PR TITLE
Migrate Flutter gallery test to null safety

### LINKED ISSUE #66395: Migrate Flutter Gallery test to NNBD
Tracks the migration to NNBD, which includes all the dependencies.

## Dependencies

Dependency | Current version | Owner | Issue/PR
-- |-- | -- | -- 
connectivity | 0.4.9+2 | Flutter Ecosystem
connectivity_for_web | 0.3.1+2 | Flutter Ecosystem
connectivity_macos | 0.1.0+4 | Flutter Ecosystem
connectivity_platform_interface | 1.0.6 | Flutter Ecosystem
device_info | 0.4.2+7 | Flutter Ecosystem
device_info_platform_interface | 1.0.1 | Flutter Ecosystem
plugin_platform_interface | 1.0.2 | Flutter Ecosystem | https://github.com/flutter/plugins/pull/3115
url_launcher | 5.5.3 | Flutter Ecosystem
url_launcher_linux | 0.0.1+1 | Flutter Ecosystem
url_launcher_macos | 0.0.1+7 | Flutter Ecosystem
url_launcher_platform_interface | 1.0.8 | Flutter Ecosystem
url_launcher_web | 1.3+2 | Flutter Ecosystem 
video_player | 0.10.6 | @blasten | https://github.com/flutter/plugins/pull/3045
video_player_platform_interface | 1.0.5 | @blasten | https://github.com/flutter/plugins/pull/3045
video_player_web | 0.1.2+3 | Flutter Ecosystem
flutter | <flutter repo> | Flutter Framework | https://github.com/dart-lang/sdk/issues/40146
flutter_web_plugins | <flutter repo> | @ditman | ~~https://github.com/flutter/flutter/issues/69843~~
intl | 0.16.1 | Dart SDK
pub_semver | 1.4.4 | Dart SDK
platform_detect | 1.4.0 | @ditman | ~~https://github.com/flutter/flutter/issues/66385~~
scoped_model | 1.0.1 | 3P
pigeon | 0.1.10 | @gaaclarke | https://github.com/flutter/flutter/issues/59118

/cc @pcsosinski

### PR DESCRIPTION
**Pull request by blasten**
Fixes https://github.com/flutter/flutter/issues/66395

### CODE REVIEW

**Comment by blasten** [comment]
@jonahwilliams I'm looking at merging this PR, which requires passing the `--enable-experiment=non-nullable` flag to the tool. Any thoughts?

**Comment by jonahwilliams** [comment]
You'll need to either wait for the flag flip when --enable-experiment is no longer necessary, or go through all of the integration tests and provide the flag manually. My understanding is that the flag flip is happening fairly soon, and this change does not need to be landed before that point.

@leafpetersen or @jakemac53 to confirm?

**Comment by jakemac53** [comment]
We are actively working on unblocking the flag flip so yes I would probably just wait if that is possible.

**Comment by blasten** [comment]
>  or go through all of the integration tests and provide the flag manually

Do you mean the Flutter gallery ones only? @jonahwilliams

**Comment by blasten** [comment]
Alright. Sounds good.

**Comment by pcsosinski** [comment]
gallery performance is a gate for being beta ready, so ideally we have this data asap

**Comment by jonahwilliams** [comment]
There are probably a hundred or so places that need --enable-experiment=non-nullable. Unless the beta plan is to ship the very first commit after the flag flip, waiting seems like a safe option to me.

**Comment by pcsosinski** [comment]
talked to @leafpetersen and @a-siva this morning, waiting for the flag flip PR to land is the plan.

**Comment by jonahwilliams** [comment]
The test scripts themselves should not be migrated to null safety, otherwise you will get sound mode warnings about the test script dependencies. This is unimportant as we care about the app under test running in sound mode.

**Comment by pcsosinski** [comment]
the flag flip has landed, what's left to get this in?

**Comment by jonahwilliams** [comment]
Looks like this needs to be rebased

**Comment by pcsosinski** [comment]
@blasten are you able to rebase this PR?

**Comment by blasten** [comment]
@pcsosinski Done

**Review by jonahwilliams — COMMENTED** [review]
You need to update the SDK constraint here:

https://github.com/flutter/flutter/blob/master/dev/integration_tests/flutter_gallery/pubspec.yaml#L5

I think it needs to be at least 2.12 to opt into null safety

**Comment by blasten** [comment]
@jonahwilliams  I followed https://dart.dev/null-safety#configure-the-sdk-version

**Comment by blasten** [comment]
@jonahwilliams have you seen this issue: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8864598928325667504/+/steps/prepare_environment/0/steps/Build_gallery/0/steps/build_gallery/0/stdout?format=raw ?

**Comment by jonahwilliams** [comment]
It looks like the versions of those plugins that were published only support the tech preview version of null safety and not the beta SDK constraint we'd need to run this without the experiment flag

**Comment by jonahwilliams** [comment]
We'd need new versions of all the plugins published with updated constraints

**Comment by tvolkert** [comment]
Seems to be blocked on https://github.com/flutter/flutter/issues/69771 and https://github.com/flutter/flutter/issues/69770 - correct?

**Comment by blasten** [comment]
I see. In the meanwhile, I will update the Dart SDK constraint for the plugins.

File: dev/integration_tests/flutter_gallery/analysis_options.yaml [hunk 1]
```diff
@@ -6,3 +6,5 @@ include: ../../../analysis_options.yaml
 analyzer:
   exclude:
     - build/**
+  enable-experiment:
```
Comment (jonahwilliams) [inline dev/integration_tests/flutter_gallery/analysis_options.yaml:9]: With the dart SDK constraint set to `>=2.12.0-0 < 3` this shouldn't be necessary.

File: dev/integration_tests/flutter_gallery/lib/demo/material/buttons_demo.dart [hunk 2]
```diff
@@ -250,7 +250,7 @@ class _ButtonsDemoState extends State<ButtonsDemo> {
               OutlinedButton.icon(
                 icon: const Icon(Icons.add, size: 18.0),
                 label: const Text('DISABLED', semanticsLabel: 'DISABLED BUTTON 6'),
-                onPressed: null,
+                onPressed: () {},
```
Comment (jonahwilliams) [inline dev/integration_tests/flutter_gallery/lib/demo/material/buttons_demo.dart:253]: I think this should be `null` so that the button stays disabled.

**Review by jonahwilliams — APPROVED** [review]
Overal changes LGTM

There is one more failure in the customer testing shard due to a constraint on intl ^0.16 which does not grab the 0.17 version since the overall version is < 1.0.

File: dev/integration_tests/flutter_gallery/lib/demo/material/buttons_demo.dart (hunk 2 above)
Comment (blasten) [reply dev/integration_tests/flutter_gallery/lib/demo/material/buttons_demo.dart]: Done

**Comment by jonahwilliams** [comment]
Fix for the font loader failure is https://github.com/flutter/engine/pull/22370

**Comment by blasten** [comment]
@jonahwilliams I sent https://github.com/flutter/engine/pull/22373. Should it be `void` or `String?`?

**Comment by blasten** [comment]
The `BUILD` file for the Gallery package needs to turn null safety on, then it should just work. fyi @renyou

**Comment by mit-mit** [comment]
This was reverted in https://github.com/flutter/flutter/pull/70023

**Pull request merged**